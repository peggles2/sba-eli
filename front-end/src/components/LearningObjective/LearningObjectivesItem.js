import React from "react";
import { List } from "semantic-ui-react";

const LearningObjectivesItem = (props) => {
  return(
    <List.Item><a href={`/learning_objectives/${props.item.id}`}>{props.item.name}</a></List.Item>
  )
}

export default LearningObjectivesItem;
