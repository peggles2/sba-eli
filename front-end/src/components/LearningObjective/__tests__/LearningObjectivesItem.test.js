import React from 'react';
import { List } from 'semantic-ui-react';
import LearningObjectivesItem from '../LearningObjectivesItem';
import { shallow } from 'enzyme';

describe('LearningObjectivesItem', () => {
  it('should render correctly', () => {
    const item = { id: 1, name: "LO 1" }
    const wrapper = shallow(<LearningObjectivesItem item={item} />); 
    
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a <List.Item>', () => {
    const item = { id: 1, name: "LO 1" }
    const wrapper = shallow(<LearningObjectivesItem item={item} />); 

    expect(wrapper.find(List.Item).length).toEqual(1);
  });
});
