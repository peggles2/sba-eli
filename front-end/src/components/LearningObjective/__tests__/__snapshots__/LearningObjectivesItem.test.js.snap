// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningObjectivesItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningObjectivesItem
    item={
      Object {
        "id": 1,
        "name": "LO 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <a
        href="/learning_objectives/1"
      >
        LO 1
      </a>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "LO 1",
        "href": "/learning_objectives/1",
      },
      "ref": null,
      "rendered": "LO 1",
      "type": "a",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <a
          href="/learning_objectives/1"
        >
          LO 1
        </a>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LO 1",
          "href": "/learning_objectives/1",
        },
        "ref": null,
        "rendered": "LO 1",
        "type": "a",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
